### ISSUE TITLE
js_of_ocaml-compiler.4.0.0 won't install with non-default opam directory

### ISSUE THREAD

**Issue by ChrisVine** [ocsigen/js_of_ocaml#1242]
**Describe the bug**
js_of_ocaml-compiler.4.0.0 installs OK with opam at its default $HOME/.opam location, but fails to install with a non-default opam location.  With an opam directory of /store/opam, I obtain the following error:

```
[ERROR] The compilation of js_of_ocaml-compiler.4.0.0 failed at "dune install -p js_of_ocaml-compiler --create-install-files js_of_ocaml-compiler".

#=== ERROR while compiling js_of_ocaml-compiler.4.0.0 =========================#
# context     2.1.2 | linux/x86_64 | ocaml-base-compiler.4.13.1 | https://opam.ocaml.org#3a74916d
# path        /store/opam/4.13/.opam-switch/build/js_of_ocaml-compiler.4.0.0
# command     /store/opam/opam-init/hooks/sandbox.sh build dune install -p js_of_ocaml-compiler --create-install-files js_of_ocaml-compiler
# exit-code   1
# env-file    /store/opam/log/js_of_ocaml-compiler-86224-9dadf8.env
# output-file /store/opam/log/js_of_ocaml-compiler-86224-9dadf8.out
### output ###
# /usr/bin/opam var prefix > /opam-tmp/dune_cb2e23_output
# [ERROR] /store/opam exists, but does not appear to be a valid opam root. Please remove it and use `opam init', or specify a different `--root' argument
```
Permissions are fine, /store/opam is a valid root and other packages install there without problems, including  js_of_ocaml-3.11.0.

**Expected behavior**
I expect js_of_ocaml-4.0.0 to install correctly

**Versions**
Version of packages used to reproduce the bug
ocaml-4.13.1, opam-2.1.2, dune-3.0.2, js_of_ocaml-4.0.0

**Comment by hhugo** [comment]
I don't think it's a jsoo issues. I would raise the issue to dune as the command that is failing is `dune install`

**Comment by ChrisVine** [comment]
This seems erroneous:

```
/usr/bin/opam var prefix > /opam-tmp/dune_cb2e23_output
```
because there are no permissions for creating a /opam-tmp directory.  Before I trouble the dune maintainers, can you confirm that a dune thing?  (As I say, if I pin to js_of_ocaml-3.11.0 then everything is fine, as are all other packages.)

**Comment by hhugo** [comment]
jsoo 4.0 uses opam file generated by dune (3.11 was not).
The generated opam file instruct to do `dune install` (3.11 did not).

I've seen multiple issues related to dune3 it looks like this is one more.

**Comment by hhugo** [comment]
This should be fixed now that dune 3.2 has been released

**Comment by hhugo** [comment]
Please reopen an issue if you still have issues with dune.3.2
